### PR TITLE
fix: Add exporter

### PR DESCRIPTION
**Pull request by nahsi**
```
# HELP gelato_block_height The latest block height
# TYPE gelato_block_height gauge
gelato_block_height 836.0
# HELP gelato_transaction_success Indicates if the last transaction was successful
# TYPE gelato_transaction_success gauge
gelato_transaction_success 1.0
# HELP gelato_balance Balance of Ethereum addresses
# TYPE gelato_balance gauge
gelato_balance{address="0x9262768725134b7d52041FAb42287EA2147D0c77",name="foobar"} 8.888887999088855e+06
```

If private key is specified it will also send some FLT from that key to itself by default once in a minute
```
2024-08-19 09:22:33,359 INFO Addresses file found: /app/addresses.yml
2024-08-19 09:22:33,444 INFO Successfully connected to the RPC endpoint: https://rpc.testnet.fluence.dev
2024-08-19 09:22:34,254 INFO Transaction successful: 0xf2e107f4164f0e2190956e14c0d9ad014aa3d923b32bc9473553620a0edfeaf6 with estimated gas: 714020
2024-08-19 09:23:34,652 INFO Transaction successful: 0x2fee74bae53714a6c6ed2506b6c77ff8351a8942319a6f4805635bf6c3d33a54 with estimated gas: 714020
```

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    if addresses_file:
+        if os.path.exists(addresses_file):
+            logger.info(f"Addresses file found: {addresses_file}")
+            with open(addresses_file, 'r') as file:
+                try:
+                    config_data = yaml.safe_load(file)
+                    config = ConfigSchema(**config_data)
+                    addresses_to_monitor = config.addresses or []
+                except (ValidationError, yaml.YAMLError) as e:
+                    logger.error(f"Error loading or validating the addresses file: {e}. Exiting.")
```
Comment (folex) [inline app.py:56]:
```suggestion
                    logger.error(f"Error loading or validating the addresses file {addresses_file}: {e}. Exiting.")
```

File: app.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        signed_txn = web3.eth.account.sign_transaction(transaction, private_key)
+        tx_hash = web3.eth.send_raw_transaction(signed_txn.rawTransaction)
+        receipt = web3.eth.wait_for_transaction_receipt(tx_hash)
+
+        if receipt.status == 1:
+            logger.info(f"Transaction successful: {tx_hash.hex()} with estimated gas: {gas}")
+            return True
+        else:
+            logger.error(f"Transaction failed: {tx_hash.hex()}")
```
Comment (folex) [inline app.py:107]: Need that data to debug

```suggestion
            logger.error(f"Transaction {tx_hash.hex()} failed: {transaction}")
```

File: app.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        transaction = {
+            'from': account.address,
+            'to': account.address,  # Sending to self
+            'value': web3.to_wei(0.01, 'ether'),
+            'nonce': nonce,
+            'chainId': chain_id
+        }
+
+        # Estimate gas cost
+        gas = web3.eth.estimate_gas(transaction)
```
Comment (folex) [inline app.py:95]: This line will fail if there's not enough balance or address is incorrect or something. Better wrap with `try except`.

Comment (folex) [reply app.py:95]: `estimateGas` is the place to catch tx errors before sending them to chain. A kind of simulation.

File: app.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+rpc_url, addresses_to_monitor, private_key, transaction_interval, port = initialize()
+web3 = connect_rpc(rpc_url)
+
+app = Flask(__name__)
+
+# Initialize Prometheus metrics
+block_height_metric = Gauge('gelato_block_height', 'The latest block height', registry=registry)
+
+def collect_metrics():
+    latest_block_height = web3.eth.block_number
```
Comment (folex) [inline app.py:138]: This way you measure block number only on request. 

Would be safer to measure inside `task` on a fixed interval. Same with balances. That way data is collected with a constant interval, and it's easier to monitor dynamics. In general. Here it's not super important I guess, so on your preference.

File: app.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    block_height_metric.set(latest_block_height)
+
+    if addresses_to_monitor:
+        balance_metrics = Gauge('gelato_balance', 'Balance of Ethereum addresses', ['address', 'name'], registry=registry)
```
Comment (folex) [inline app.py:142]: Is it OK to recreate Gauge every time? 🤔 Usually they are reused as they collect data inside. Not sure how it's implemented here, but I'd better move all `*_metrics` to global variables.

**Review by folex — APPROVED** [review]
Nice!

File: app.py (hunk 4 above)
Comment (nahsi) [reply app.py:138]: I wanted to have a correlation between value and the timestamp in Prometheus so we can tell that at the timestamp of the metric the value was N. I'm not if we need that though.
Also scrape_interval of prometheus makes it kinda fixed.

Let's go with the way it is right now and later I will change to a fixed interval if some issues occur.

**Pull request merged**